src/server/shell/abstract_shell.cpp

```diff
@@ -247,15 +250,21 @@ auto msh::AbstractShell::create_surface(
     // Instead of a shared pointer, a local variable could be used and the lambda could capture a reference to it
     // This should be safe, but could be the source of nasty bugs and crashes if the wm did something unexpected
     auto const should_decorate = std::make_shared<bool>(false);
-    auto const build = [observer, observer_executor, should_decorate, wayland_surface](
+    auto const build = [observer, observer_executor, should_decorate, wayland_surface, this](
```

Comment:
As part of this refactor we should be losing the `wayland_surface` parameter to `Shell::create_surface()` but we are still using it here.